proofs.go

```diff
@@ -164,6 +164,8 @@ type ProofD struct {
 	cachedRangeStructures map[int][]*rangeproof.ProofStructure
 }
 
+// MergeProofP merges a ProofP into the ProofD. The public key is only passed for consistency with
+// ProofU's MergeProofP method.
 func (p *ProofD) MergeProofP(proofP *ProofP, pk *gabikeys.PublicKey) {
```

Comment:
* Although the reason specified in this comment for including the public key parameter is not incorrect, it would be better to say here that we pass it to comply with the `Proof` interface, which has a `MergeProofP(*ProofP, *gabikeys.PublicKey)` method.
* In case a parameter is not used like here, it is best to not name it but write `_` instead of a name. That makes it explicit that this parameter has no actual name.